### PR TITLE
feat(dev): Upgrade zookeeper/kafka dev containers for Intel hosts

### PR DESCRIPTION
**Pull request by armenzg**
In #28574, we upgraded the versions of zookeeper/kafka to 6.x only for Apple
arm64 hosts.

When trying to use such version on Intel, it would fail because version 6.x
requires specifying the path to `data` *absolutely* (see [docs][docs]), otherwise, you would get this
error:

> Command [/usr/local/bin/dub path /var/lib/kafka/data writable] FAILED !

If you don't rename the key of the moount bind, users with an existing set-up would not get
the absolue path properly configured, thus, they would still face the issue (see [comment][comment]).
In other words, people starting from scratch would have no issues.

At the time, we decided to back things out and investigate later.

This change reunites Intel and arm64 hosts.

[docs]: https://docs.confluent.io/platform/current/installation/docker/operations/external-volumes.html#data-volumes-for-kafka-and-zk
[comment]: https://github.com/getsentry/sentry/pull/28574#issuecomment-922087892

### CODE REVIEW

**Review by BYK — APPROVED** [review]
We are not using v6 for on-premise so kind of blind-accepting.

FYI, with this version, we can drop zookeeper:

- https://www.confluent.io/blog/kafka-without-zookeeper-a-sneak-peek/
- https://github.com/apache/kafka/blob/6d1d68617ecd023b787f54aafc24a4232663428d/config/kraft/README.md

**Comment by armenzg** [comment]
> We are not using v6 for on-premise so kind of blind-accepting.
> 
> FYI, with this version, we can drop zookeeper:
> 
>     * https://www.confluent.io/blog/kafka-without-zookeeper-a-sneak-peek/
> 
>     * https://github.com/apache/kafka/blob/6d1d68617ecd023b787f54aafc24a4232663428d/config/kraft/README.md

I'm aware of that. I've added it to the backlog.
